libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.m

```diff
@@ -164,133 +94,145 @@ - (void)setCustomRequestBodyStream:(NSInputStream* (^)(void))bodyInputStreamBloc
     }
 }
 
-#pragma mark - Custom headers
-
-- (NSDictionary *)customHeaders
-{
-    return self.action.allHTTPHeaderFields;
-}
-
-- (void)setCustomHeaders:(NSDictionary *)customHeaders
-{
-    for (NSString *key in [customHeaders allKeys]) {
-        [self setHeaderValue:customHeaders[key] forHeaderName:key];
-    }
-}
-
-- (void)setHeaderValue:(NSString *)value forHeaderName:(NSString *)name
-{
-    if (name == nil)
-        return;
-    
-    if ([value length] > 0) {
-        [self.action setValue:value forHTTPHeaderField:name];
-    } else {
-        // TODO: Network SDK doesn't have a header removal mechanism at this point.
-    }
-}
-
 # pragma mark - send and cancel
 
-- (void)prepareRequestForSend
-{
-    // We need to do some jujitsu to figure out how best to form up the request, based on the way the
-    // request is configured.
-    
-    // Sanity check the path against the endpoint value.
-    if (self.endpoint.length > 0 && [self.path hasPrefix:self.endpoint]) {
-        self.path = [self.path substringFromIndex:self.endpoint.length];
-    }
-    
-    // Custom request body overrides default behavior.
-    if (self.requestBodyStreamBlock != nil) {
-        self.action.parameters.bodyStreamBlock = self.requestBodyStreamBlock;
-        if ([self.requestContentType length] > 0) {
-            [self setHeaderValue:self.requestContentType forHeaderName:@"Content-Type"];
+- (NSURLRequest *)prepareRequestForSend {
```

Comment:
Guts of forming the request properly are done here, including handling of communities, etc.